proxy/handlers/events.go

```diff
@@ -47,3 +60,16 @@ func (h *eventsHandler) pushEvents(c *gin.Context) {
 
 	JsonResponse(c, http.StatusOK, nil, "")
 }
+
+func (h *eventsHandler) createMiddleware() []gin.HandlerFunc {
+	var middleware []gin.HandlerFunc
+
+	if h.config.Username != "" && h.config.Password != "" {
```

Comment:
so the username and the password must be the same as the ones in config file, right? for example, on proxy we hash the received username and password and compare them with the ones from config file

Reply:
exactly. the factory creates an httpClient which will require these fields, so it can set auth headers